### PR TITLE
Clean up MemorySanitizer library warnings

### PR DESCRIPTION
**Pull request by divergentdave**
This modifies the `patch_build.py` script so that it won't issue spurious warnings about standard libraries for which interceptors are provided. I got the list of libraries from the discussion on #1725.

I also updated `infra/base-images/all.sh`, as it was missing two of the base containers, and I needed to rebuild one of them locally to test.

### CODE REVIEW

**Comment by divergentdave** [comment]
It appears that the build is failing because the python virtualenv used in the "Infra tests" workflow does not have python-apt installed. Since this PR touches infra/base-images/base-sanitizer-libs-builder/patch_build.py, the tests inside infra/base-images/base-sanitizer-libs-builder/msan_build.py are getting conditionally run, and failing to import that module. It appears the tests in this directory haven't run since the CI was overhauled recently. The python-apt package on PyPI is broken and out of date, while the system libraries are built for the system python version, not 3.8, and they also won't work. It appears the only remaining option is to build python-apt from source.

**Comment by divergentdave** [comment]
Well, that took me far afield from the original purpose of this PR, but at least I got the build working on a couple more directories. Let me know if you'd like me to dice this up into separate branches.

**Review by oliverchang — CHANGES_REQUESTED** [review]
Thank you for the various lint and formatting fixes! Left some minor comments.

Unfortunately that does make it a litle difficult to figure out exactly what functionality changed. As far as I can tell, it's mostly the addition of `patch_build.INTERCEPTED_LIBRARIES`. Is there anythign else?

File: infra/base-images/base-sanitizer-libs-builder/packages/openssl.py [hunk 1]
```diff
@@ -14,29 +14,28 @@
...
+def add_no_asm_arg(config_path):
   """Add --no-asm to config scripts."""
   shutil.move(config_path, config_path + '.real')
-  with open(config_path, 'w') as f:
-    f.write(
-        '#!/bin/sh\n'
-        '%s.real no-asm "$@"\n' % config_path)
-  os.chmod(config_path, 0755)
+  with open(config_path, 'w') as config_file:
+    config_file.write('#!/bin/sh\n' '%s.real no-asm "$@"\n' % config_path)
```
[the remark this answers is not in the document]
Comment (divergentdave) [reply infra/base-images/base-sanitizer-libs-builder/packages/openssl.py]: Yes, it was. I added a backslash to escape the newline in the source code, and that keeps pylint happy.

**Comment by divergentdave** [comment]
Yes, adding the list of intercepted libraries was the original impetus for this PR, and the only functional change. The commit before that was to make it easier to rebuild the relevant image locally, and all the commits after were for getting the CI working again. (Formatting, missing license header, docstrings, naming conventions, a missing dependency, and incorrect test cases in compiler_wrapper_test.py)

**Review by oliverchang — APPROVED** [review]
Thank you! Looks great to me. I'll hold off on merging this until next week though, as it's end of the week for most people due to holidays.

**Comment by divergentdave** [comment]
I see this was all backed out, presumably due to an issue with the renaming I did. Would you accept a cherry-pick of 96a54fe48f6cd158b9d3aff1461f5612cd7ff5f2 that failed pylint?

**Pull request merged**